brave/brave-core
# Download Brave Wallet component on demand on Android

Resolves https://github.com/brave/brave-browser/issues/31824

This PR does 3 things:
1. Prevents automatic registration of `Brave Wallet data files` component for Android (`wallet_data_files_installer.cc`);
2. Performs on demand registration of the component after user presses `Get started`/`Restore` button during Wallet onboarding (`SetupWalletFragment.onViewCreated`)
3. Does display of the progress of the component download during onboarding screens (`DownloadComponentProgressFragment.java`)

The PR doesn't do anything with unregister component and cleanup downloaded/unpacked fiels.

Screenshots:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Description &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|Screenshot
---|---
`Secure your crypto` screen|![1-1](https://github.com/brave/brave-core/assets/24739341/d158d498-4ca0-4890-9f37-78e43048a30a)
`Backup your wallet` screen|![1-2](https://github.com/brave/brave-core/assets/24739341/0c09df3d-ae3e-4aa9-a9b6-5cf3530b751e)
`Your recovery phrase` screen|![1-3](https://github.com/brave/brave-core/assets/24739341/c64658e9-cc00-409e-bb28-5ea0818d4695)
`Verify recovery phrase`|![1-4](https://github.com/brave/brave-core/assets/24739341/f175c612-38e8-460a-b488-7af75d2d9532)
`Restore crypto account`|![2-1](https://github.com/brave/brave-core/assets/24739341/0cb5891f-77df-4999-99b3-29bb0f2b61e8)




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
I. Normal flow, unlimited fast internet connection
1. Install Brave
2. Open brave://components, ensure there is no `Brave Wallet data files` component
3. Open Wallet => Get Started => ... you must not see or see for a moment the `Download wallet data %` label
4. Open brave://components, ensure there is no `Brave Wallet data files` component is in list with Up-to-data status.

II. Create wallet flow on slow internet connection
1. Configure internet speed of the device to be 256 kbit/s, it can be done through WiFi router settings
2. Install Brave
3. Open Wallet => `Get Started`. Expected to have `Download wallet data file` on `Secure your crypto` / `Backup your wallet` / `Your recovery phrase` / `Verify recovery phrase` screens

III. Recover wallet flow on slow internet connection
1. Configure internet speed of the device to be 256 kbit/s, it can be done through WiFi router settings
2. Install Brave
3. Open Wallet => Restore.  Expected to have `Download wallet data file` on `Restore crypto account` screen.
